src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/getUsers';
+import { Todo } from '../../types/todo';
+
+type Props = {
+  onSubmit: (todos: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handlTitleChange = (
+    inputTitleEvent: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTitle(inputTitleEvent.target.value);
+    setHasTitleError(false);
+  };
+
+  const handlUserIdChange = (
+    inputUserEvent: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setUserId(+inputUserEvent.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handlSubmit = (eventSubmit: React.FormEvent) => {
+    eventSubmit.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title: title,
+      completed: false,
+      userId: userId,
+      user: getUserById(userId),
+    });
+
+    reset();
+  };
```

Comment:
[CODE KNOWLEDGE] - While handling form values, trigger submit logic using `onSubmit` prop of form, instead of `onClick` submit button. You are correctly using the `onSubmit` prop of the form to handle the form submission. This is the correct way to handle form submissions in React as it allows the form to be submitted by both clicking the submit button and pressing the enter key.